src/components/pager/pager.component.js

```diff
@@ -121,27 +123,34 @@ const Pager = ({
 
   const handleOnPagination = useCallback(
     (e) => {
+      setValue(e.target.value);
       setCurrentPageSize(Number(e.target.value));
       onPagination(1, Number(e.target.value), "page-select");
     },
     [onPagination]
   );
 
+  const handleKeyDown = useCallback(
+    (e) => Events.isEnterKey(e) && handleOnPagination(e),
+    [handleOnPagination]
+  );
+
   const sizeSelector = () => {
     return (
       <StyledSelect
-        value={String(currentPageSize)}
-        onChange={(ev) => {
-          handleOnPagination(ev);
-        }}
+        value={String(value)}
+        onChange={(ev) => setValue(ev.target.value)}
+        onKeyDown={handleKeyDown}
         data-element="page-select"
         id="page-select"
       >
         {pageSizeSelectionOptions.map((sizeOption) => (
           <Option
+            type="button"
```

Comment:
These options aren't buttons so this shouldn't be here

Reply:
I thought that from accessibility point of view it can be good to mark it as a type="button"